packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -34,43 +36,99 @@ describe("<DateRangeInput>", () => {
         expect(inputRef.firstCall.args[0]).to.be.an.instanceOf(HTMLInputElement);
     });
 
-    it("invokes onChange when a day is selected or deselected in the picker", () => {
-        const onChange = sinon.spy();
-        const { root, getDayElement } = wrap(<DateRangeInput onChange={onChange} />);
-        root.setState({ isOpen: true });
-
-        // select days
-        getDayElement(22).simulate("click");
-        getDayElement(24).simulate("click");
-
-        // deselect days
-        getDayElement(22).simulate("click");
-        getDayElement(24).simulate("click");
-
-        // TODO: Make these checks more rigorous once controlled mode is supported,
-        // when we'll be able to enforce which initial month is shown.
-        expect(onChange.callCount).to.deep.equal(4);
-    });
-
     it("shows empty fields when no date range is selected", () => {
-        const onChange = sinon.spy();
-        const { root } = wrap(<DateRangeInput onChange={onChange} />);
+        const { root } = wrap(<DateRangeInput />);
 
         expect(getStartInputText(root)).to.be.empty;
         expect(getEndInputText(root)).to.be.empty;
     });
 
-    it("shows formatted dates in fields when date range is selected", () => {
-        const onChange = sinon.spy();
-        const { root, getDayElement } = wrap(<DateRangeInput onChange={onChange} />);
-        root.setState({ isOpen: true });
+    describe("when uncontrolled", () => {
+        it("Clicking a date puts the selected date range in the input boxes", () => {
+            const defaultValue = [new Date(2017, Months.JANUARY, 22), null] as DateRange;
+
+            const { root, getDayElement } = wrap(<DateRangeInput defaultValue={defaultValue} />);
+            root.setState({ isOpen: true });
+
+            getDayElement(24).simulate("click");
 
-        getDayElement(22).simulate("click");
-        getDayElement(24).simulate("click");
+            expect(getStartInputText(root)).to.equal("2017-01-22");
+            expect(getEndInputText(root)).to.equal("2017-01-24");
+        });
 
-        // TODO: Make these checks more rigorous once controlled mode is supported.
-        expect(getStartInputText(root)).to.be.not.empty;
-        expect(getEndInputText(root)).to.be.not.empty;
+        it("Clearing the date range clears the inputs, and invokes onChange with [null, null]", () => {
```

Comment:
Dunno if it's already on your radar, but this reminded me of the `allowSingleDayRange` from DateRangePicker. Not sure if necessary to support.